packages/hoppscotch-app/components/http/Body.vue

```diff
@@ -80,19 +91,45 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from "@nuxtjs/composition-api"
+<script setup lang="ts">
+import { Ref, watch, ref } from "@nuxtjs/composition-api"
+import { HoppRESTHeader } from "@hoppscotch/data"
+import { RequestOptionTabs } from "./RequestOptions.vue"
 import { useStream } from "~/helpers/utils/composables"
-import { restContentType$, setRESTContentType } from "~/newstore/RESTSession"
 import { knownContentTypes } from "~/helpers/utils/contenttypes"
+import {
+  restContentType$,
+  restHeaders$,
+  setRESTContentType,
+  setRESTHeaders,
+} from "~/newstore/RESTSession"
 
-export default defineComponent({
-  setup() {
-    return {
-      validContentTypes: Object.keys(knownContentTypes),
+const emit = defineEmits(["changeTab"])
```

Comment:
Vue for TypeScript has an alternative way to define emits and props which tell the proper type signature of the emit functions.
Please check out: https://vuejs.org/api/sfc-script-setup.html#typescript-only-features